### DISCUSSION TITLE
Recursive queries

### DISCUSSION THREAD

**Discussion by ghost** [sqlc-dev/sqlc#718] [Q&A] [answered]
Hi thanks for this great tool.

Is there any support for recursive queries? I have some tables that are hierarchical, and I need to flatten them out. 
It's no problem for me to do this part by hand, but i'm just wondering if this tool has any support for it.

**Comment by kyleconroy** [comment, marked as answer]
Yes! sqlc can handle all types of queries. If you have a specific query in mind, feel free to write that query out (don't forget the table definition). You can also play around with sqlc in the browser over at https://play.sqlc.dev/.

**Reply by ghost** [reply]
Surprisingly this ended up working fine. Thanks cool tool. Switched off my ORM for it.

**Reply by basilex** [reply]
> Yes! sqlc can handle all types of queries. If you have a specific query in mind, feel free to write that query out (don't forget the table definition). You can also play around with sqlc in the browser over at https://play.sqlc.dev/.

See here: https://github.com/kyleconroy/sqlc/issues/1021
But recursive queries works fine. Thanx a lot for your effort!!
